chore(fake-driver): annihilate loathsome mixin classes

boneskull opened the pull request:
I did here what I did to `appium-xcuitest-driver`, which was to assign the command methods to the class as fields.  Easier this way.

Also, it's more difficult to parse for automatic command docs using the old way.

boneskull committed:
chore(fake-driver): annihilate loathsome mixin classes

I did here what I did to `appium-xcuitest-driver`, which was to assign the command methods to the class as fields.  Easier this way.

boneskull commented:
Current dependencies on/for this PR:
* master
  * **PR #18058** <a href="https://app.graphite.dev/github/pr/appium/appium/18058" target="_blank"><img src="https://static.graphite.dev/graphite-32x32.png" alt="Graphite" width="10px" height="10px"/></a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/appium/appium/18058?utm_source=stack-comment).

jlipps commented on packages/fake-driver/lib/commands/alert.js:
```diff
@@ -1,82 +1,73 @@
- * @returns {Class<AlertCommands>}
- */
-export function AlertMixin(Base) {
+export default {
```
why `export default` instead of `export async function` for each one?

boneskull replied:
six and 1/2 dozen I guess

boneskull replied:
@jlipps I could go either way here. given these aren't intended to be imported by anything other than `commands/index.js`, if you have a preference I can change it.

boneskull replied:
I'll change it later if you wish

mykola-mokhnach commented:
LGTM, but tests seem to be unhappy

boneskull committed:
chore(scripts): fix typo in sync-pkgs script

The typo caused the `keywords` field in `packages/typedoc-plugin-appium/package.json` to be overwritten.  Because of this, the `keywords` field lost the magic keyword that `typedoc` uses to automatically detect a plugin.  Thus, `@appium/typedoc-plugin-appium` could not be found, and its specific options caused `typedoc` to fail.

boneskull committed:
fix(typedoc-plugin-appium): fix keywords for typedoc plugin autodetection

boneskull committed:
fix(typedoc-plugin-appium): reduce severity of "missing method" error

This is either a bug in the plugin or a bug in `@appium/base-driver` (tbd).  But it was causing `typedoc` to fail with a fatal error, and we don't want that to happen (yet).  This change punts on the problem to get the docs build working

boneskull committed:
fix(docs): fix filename of basedriver commands

I'm unsure about a) where the filename I just changed came from, and b) if we have control over the filename, and c) if so, how much control.

But anyway: it seems to generate a file named `appium_base_driver-1._appium_base_driver.md`.

boneskull committed:
chore(typedoc-plugin-appium): reorganize the tests

many of these were actually E2E tests (and slow)

boneskull committed:
chore(typedoc-plugin-appium): remove cruft

boneskull committed:
fix(typedoc-plugin-appium): non-appium theme should not cause a fatal error

This was preventing tests from running

boneskull committed:
chore(typedoc-plugin-appium): disable sorting

Sorting is handled by typedoc and it supersedes whatever we do with it.  The correct way to implement a custom sort would be...something else.  YAGNI for now
